econml/score/rscorer.py

```diff
@@ -0,0 +1,219 @@
+        How to aggregate the nuisance value for each sample across the `mc_iters` monte carlo iterations of
+        cross-fitting.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+    """
+
+    def __init__(self, *,
+                 model_y,
+                 model_t,
+                 discrete_treatment=False,
+                 categories='auto',
+                 n_splits=2,
+                 mc_iters=None,
+                 mc_agg='mean',
+                 random_state=None):
+        self.model_y = clone(model_y, safe=False)
+        self.model_t = clone(model_t, safe=False)
+        self.discrete_treatment = discrete_treatment
+        self.n_splits = n_splits
+        self.categories = categories
+        self.random_state = random_state
+        self.mc_iters = mc_iters
+        self.mc_agg = mc_agg
+
+    def fit(self, y, T, X=None, W=None, sample_weight=None, groups=None):
+        """
+
+        Parameters
+        ----------
+        Y: (n × d_y) matrix or vector of length n
+            Outcomes for each sample
+        T: (n × dₜ) matrix or vector of length n
+            Treatments for each sample
+        X: optional (n × dₓ) matrix
+            Features for each sample
+        W: optional (n × d_w) matrix
+            Controls for each sample
+        sample_weight: optional (n,) vector
+            Weights for each row
+        groups: (n,) vector, optional
+            All rows corresponding to the same group will be kept together during splitting.
+            If groups is not None, the n_splits argument passed to this class's initializer
+            must support a 'groups' argument to its split method.
+
+        Returns
+        -------
+        self
+        """
+        if X is None:
+            raise ValueError("X cannot be None for the RScorer!")
+
+        self.lineardml_ = LinearDML(model_y=self.model_y,
+                                    model_t=self.model_t,
+                                    n_splits=self.n_splits,
+                                    discrete_treatment=self.discrete_treatment,
+                                    categories=self.categories,
+                                    random_state=self.random_state,
+                                    mc_iters=self.mc_iters,
+                                    mc_agg=self.mc_agg)
+        self.lineardml_.fit(y, T, X=None, W=np.hstack([v for v in [X, W] if v is not None]),
+                            sample_weight=sample_weight, groups=groups, cache_values=True)
+        self.base_score_ = self.lineardml_.score_

```

Comment:
seems the score function under `_ortho_learner` needs to be updated to incorporate Monte Carlo simulation?  since now we cached nuisances it's will be much easier

Reply:
what do you mean?

Reply:
I see here is fine since it's the training set score, and the score method is for a new set of data X, but still need to incorporate MC simulation?

Reply:
I don;t think we need this here. But in general we could enable the mc simulation in score of _ortho_learner, but that would require to store all iterations. This seems to be related to the TODO in ortholearner where we say that we need to potentially store all nuisance estimates from all mc iters. Currently we only store the final one. A new PR which fixes that could also fix the scoring to use all the mc iters.

Reply:
Yes exactly!

Reply:
Wait, the stored nuisances are the aggregated ones, not the final ones; it's just the model where we store only the final one (but maybe that's what you meant)